DependencyInjection/MonologExtension.php

```diff
@@ -417,32 +417,66 @@ private function buildHandler(ContainerBuilder $container, $name, array $handler
             $nestedHandlerId = $this->getHandlerId($handler['handler']);
             $this->markNestedHandler($nestedHandlerId);
 
+            $isLegacyBridge = !\class_exists('Symfony\Bridge\Monolog\Processor\SwitchUserTokenProcessor');
+
             if (isset($handler['activation_strategy'])) {
                 $activation = new Reference($handler['activation_strategy']);
             } elseif (!empty($handler['excluded_404s'])) {
                 if (class_exists(HttpCodeActivationStrategy::class)) {
                     @trigger_error('The "excluded_404s" option is deprecated in MonologBundle since version 3.4.0, you should rely on the "excluded_http_codes" option instead.', E_USER_DEPRECATED);
                 }
-                $activationDef = new Definition('Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy', [
-                    new Reference('request_stack'),
-                    $handler['excluded_404s'],
-                    $handler['action_level']
-                ]);
+                if ($isLegacyBridge) {
+                    $activationDef = new Definition('Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy', [
+                        new Reference('request_stack'),
+                        $handler['excluded_404s'],
+                        $handler['action_level']
+                    ]);
+                } else {
+                    $innerActivationDef = new Definition('Monolog\Handler\FingersCrossed\ErrorLevelActivationStrategy', [
+                        $handler['action_level']
+                    ]);
+                    $container->setDefinition($handlerId.'.error_level_strategy', $innerActivationDef);
+                    $activationDef = new Definition('Symfony\Bridge\Monolog\Handler\FingersCrossed\NotFoundActivationStrategy', [
+                        new Reference('request_stack'),
+                        $handler['excluded_404s'],
+                        new Reference($handlerId.'.error_level_strategy')
```

Comment:
imo you don't need a reference here (or even to declare the service with the strategy). Just nest the definition as it's internal stuff.